corehq/apps/reports/tasks.py

```diff
@@ -381,49 +385,24 @@ def _get_export_properties(export_id, export_is_legacy):
                         properties.add("-".join(path_parts))
     return properties
 
-def _get_attachment_dicts_from_form(form):
-    if 'external_blobs' in form:
-        return form['external_blobs'].values()
-    elif '_attachments' in form:
-        return form['_attachments'].values()
-    return []
-
 
 def _get_form_ids_having_multimedia(domain, app_id, xmlns, startdate, enddate, export_is_legacy):
     """
     Return a list of form ids.
     Each form has a multimedia attachment and meets the given filters.
     """
     if not export_is_legacy:
-        # TODO: Remove references to _attachments once all forms have been migrated to Riak
-        query = (FormES()
-                 .domain(domain)
-                 .app(app_id)
-                 .xmlns(xmlns)
-                 .submitted(gte=parse(startdate), lte=parse(enddate))
-                 .remove_default_filter("has_user")
-                 .source(['_attachments', '_id', 'external_blobs']))
-        form_ids = set()
-        for form in query.scroll():
-            try:
-                for attachment in _get_attachment_dicts_from_form(form):
-                    if attachment['content_type'] != "text/xml":
-                        form_ids.add(form['_id'])
-                        continue
-            except AttributeError:
-                pass
+        fetch_fn = es_get_form_ids_having_multimedia
     else:
-        key = [domain, app_id, xmlns]
-        form_ids = {
-            f['id'] for f in
-            XFormInstance.get_db().view(
-                "attachments/attachments",
-                start_key=key + [startdate],
-                end_key=key + [enddate, {}],
-                reduce=False
-            )
-        }
-    return form_ids
+        fetch_fn = couch_get_form_ids_having_multimedia
```

Comment:
nit but I'd swap the if logic around - always easier to read if you don't have to boolean logic in your head.